sieve-of-eratosthenes/sieve.js

```diff
@@ -17,6 +17,24 @@ function eratosthenes(limit) {
                 primes.push(p);
         }
     }
+
+    if (limit >= 2) {
+        var sqrtlmt = Math.sqrt(limit) - 2;
+        var nums = new Array(); // start with an empty Array...
+        for (var i = 2; i <= limit; i++) // and
+            nums.push(i); // only initialize the Array once...
+        for (var i = 0; i <= sqrtlmt; i++) {
+            var p = nums[i]
+            if (p)
+                for (var j = p * p - 2; j < nums.length; j += p)
```

Comment:
![Codacy](http://localhost:9000/assets/images/favicon.png) Issue found: [Variable 'j' is already declared in line 11.](http://localhost:9000/app/rtfrodrigo_3/complexity-score/file/16/issues/source?bid=2&fileBranchId=3#l29)